setuptools will disallow newlines in descriptions, so do not generate them
see e.g. https://bugs.gentoo.org/765868

Index: catkin_pkg-0.4.23/src/catkin_pkg/python_setup.py
===================================================================
--- catkin_pkg-0.4.23.orig/src/catkin_pkg/python_setup.py
+++ catkin_pkg-0.4.23/src/catkin_pkg/python_setup.py
@@ -99,9 +99,9 @@ def generate_distutils_setup(package_xml
         data['url'] = package.urls[0].url
 
     if len(package.description) <= 200:
-        data['description'] = package.description
+        data['description'] = package.description.replace('\n', ' ')
     else:
-        data['description'] = package.description[:197] + '...'
+        data['description'] = package.description[:197].replace('\n', ' ') + '...'
         data['long_description'] = package.description
 
     data['license'] = ', '.join(package.licenses)
